README.md
# To Do List
To Do List created in React Native

## **Folders Structure**
- **assets** (images)
- **components** (only reuseable part of code)
- **screens** (all apps screens (pages))

## **Usage**

1. Download and install expo app on your phone
    - Android: https://play.google.com/store/apps/details?id=host.exp.exponent&hl=pl
    - iOS: https://itunes.apple.com/us/app/expo-client/id982107779?mt=8
    - after download app you must login to your profile in app
2. Start project:
    - you need node 11 (the newest version)
    - go to root folder
    - install yarn if you do not have
    - yarn install
    - npm install expo-cli --global
    - expo start --android/ expo start --ios
    - open http://localhost:19002 - Expo DevTools
    - click 'Tunnel' in lower left corner screen
    - in termin type "?" to open expo help
    - in termin type "s" and login to your account
3. Go to app in your phone
4. Scan QR code or click on link
5. Wait for download
6. Test your app :) -> hot reload is default after every time if you save file.
7. Building apk/ipa
    - expo build:android / expo build:ios (https://docs.expo.io/versions/latest/distribution/building-standalone-apps)
    - you can see progress: https://expo.io/builds

To-Do-List/CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [1.6.2]- 2020-05-27
### Changes
fill changelog file

## [1.6.1]- 2020-05-26
### Changes
move readme file

## [1.6.0] - 2020-05-26
### Added
add folders structure to readme file

## [1.5.2]- 2020-05-26
### Added
add validation to list item

## [1.5.1] - 2020-05-26
### Added
add missing dependencies - fix issue

## [1.5.0]- 2020-05-20
### Added
add new splash screen icon

## [1.4.2] - 2020-05-12
### Added
add changelog file into project

## [1.4.1]- 2020-05-12
### Added
add styling to last page

## [1.4.0] - 2020-05-05
### Changes
improve styling on main page

## [1.3.2] - 2020-04-28
### Changes
improve way of using functions

## [1.3.1]- 2020-04-20
### Added
add image in home page
add basic list and handler to remove them
### Changes
change files from js to typestript
change styling on every page

## [1.3.0] - 2020-04-14
### Added
add icons to navigation

## [1.2.2] - 2020-04-14
### Added
custom button styling

## [1.2.1] - 2020-04-13
### Changes
separate screens

## [1.2.0] - 2020-04-01
### Added
add basic styling of navigation pages

## [1.1.2] - 2020-03-25
### Added
create basic navigation between home and settings screen

## [1.1.0] - 2020-03-04
### Changes
update readme

## [1.0.0] - 2020-03-24
### Added
initial commit

To-Do-List/App.js
import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

import Home from './components/home';
import TasksList from './components/tasksList';
import LastScreen from './components/lastScreen';
import NewTask from './components/newTask';

const HomeStack = createStackNavigator();
const TasksListStack = createStackNavigator();
const LastScreenStack = createStackNavigator();
const Tab = createBottomTabNavigator();

function HomeStackScreen() {
  return (
      <HomeStack.Navigator>
        <HomeStack.Screen name="Home" component={Home} />
      </HomeStack.Navigator>
  );
}

function TasksListStackScreen() {
  return (
    <TasksListStack.Navigator>
      <TasksListStack.Screen name="TasksList" component={TasksList} />
      <TasksListStack.Screen name="NewTask" component={NewTask} />
    </TasksListStack.Navigator>
  );
}

function LastScreenStackScreen() {
  return (
    <LastScreenStack.Navigator>
      <LastScreenStack.Screen name="LastScreen" component={LastScreen} />
    </LastScreenStack.Navigator>
  );
}

export default function App() {

  return (
    <NavigationContainer>
      <Tab.Navigator
            screenOptions={({ route }) => ({
              tabBarIcon: ({ color, size }) => {
                let iconName;
    
                if (route.name === 'Home') {
                  iconName =  'ios-home';
                } else if (route.name === 'TasksList') {
                  iconName =  'ios-list-box';
                } else if (route.name === 'LastScreen') {
                  iconName = 'ios-trophy';
                } 
    
                // You can return any component that you like here!
                return <Ionicons name={iconName} size={size} color={color} />;
              },
            })}
            tabBarOptions={{
              activeTintColor: '#191970',
              inactiveTintColor: '#798291',
            }}
          >
        <Tab.Screen name="Home" component={HomeStackScreen} />
        <Tab.Screen name="TasksList" component={TasksListStackScreen} />
        <Tab.Screen name="LastScreen" component={LastScreenStackScreen} />
      </Tab.Navigator>
    </NavigationContainer>
  );

}

To-Do-List/components/tasksList.js
import * as React from 'react';
import { TouchableOpacity, Text, View, StyleSheet } from 'react-native';

export default function TasksList({ navigation }) {

    const createNewTask = () => {
        navigation.navigate('NewTask');
    }

    return (
        <View style={styles.tasksList}>
            <Text>Tasks List screen</Text>
            <TouchableOpacity onPress={createNewTask} style={styles.buttonContainer}>
                <Text style={styles.button}>+</Text>
            </TouchableOpacity>
        </View>
    );
}

const styles = StyleSheet.create({
    tasksList: {
        flex: 1, 
        justifyContent: 'center', 
        alignItems: 'center',
        backgroundColor: '#D3C0F9',
    },
    buttonContainer: {
        width: 60,
        height: 60,
        borderRadius: 100/2,
        backgroundColor: '#fffacd',
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 4,
        },
        shadowOpacity: 0.30,
        shadowRadius: 4.65,
        elevation: 8,
        margin: 25,
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        position: 'absolute',
        bottom: 0,
        right: 0,
    },
    button: {
        color: '#798291',
        fontSize: 28,
    },
})

